### PR TITLE
Show failure in ssl-opts.sh  when key export fails

### PR DESCRIPTION
**Pull request by RonEld**
## Description
1. When `ssl_server2` export key functionality,
don't exit the server, but reset it.
2. Add text filters for `export keys functionality` test in sll-opts.sh
to check for additional output, for verifying export succeeded.

This PR will fix the false positive reported in #2662 , but it doesn't fix the root cause for the failure, which is reported in https://github.com/ARMmbed/mbed-crypto/issues/137

## Status
**READY**

## Requires Backporting
NO  
This bug is on a new feature which isn't in any LTS branch

## Additional comments
Once this PR wioll be merged, the CI will fail, until https://github.com/ARMmbed/mbed-crypto/issues/137 is fixed and merged!

## Todos
- [ ] Tests
- [ ] Documentation
- [ ] Changelog updated
- [ ] Backported

### CODE REVIEW

**Comment by RonEld** [comment]
Maybe I explained wrong, but point (2) is exactly the scenario.
The use issue is that there is a false poistive in the test, claiming the test succeeded even though it should have failed:
~~~
DTLS proxy: 3d, gnutls server .......................................... PASS
DTLS proxy: 3d, gnutls server, fragmentation ........................... PASS
DTLS proxy: 3d, gnutls server, fragmentation, nbio ..................... PASS
export keys functionality .............................................. ssl-opt.sh: 619: kill: No such process

PASS
------------------------------------------------------------------------
PASSED (641 / 641 tests (27 skipped))
~~~

File: programs/ssl/ssl_server2.c [hunk 1]
```diff
@@ -3199,7 +3199,7 @@ int main( int argc, char *argv[] )
         {
             mbedtls_printf( " failed\n  ! mbedtls_ssl_tls_prf returned -0x%x\n\n",
                             -ret );
-            goto exit;
+            goto reset;
```
Comment (gilles-peskine-arm) [inline programs/ssl/ssl_server2.c]: Why this change? I don't know if it's correct or not, I don't understand it.

Comment (gilles-peskine-arm) [reply programs/ssl/ssl_server2.c:3202]: ping @RonEld : do you remember why you made this change? The commit message says what was changed but not why.

Comment (RonEld) [reply programs/ssl/ssl_server2.c:3202]: All I remember is that the server showed false positive on the results. It passed even though it should have failed.
I don't exactly remember why I did this change, but I think it was to fix the `ssl-opt.sh: 619: kill: No such process` part of the test. the server finished without a graceful shutdown, If I remember correct.

To elaborate, there were two issues:
1. the test result was a false positive
2. the server process terminated before the test (`ssl-opts/sh`) could;d terminate it

**Comment by danh-arm** [comment]
I can't see the originally reported issue in #2662. There's no "ssl-opt.sh: 619: kill: No such process" in a recent release build job:
https://jenkins-internal.mbed.com/job/mbedtls-release-new/736

Unless anyone else can see something untoward here I propose we close this PR and the bug.

**Comment by gilles-peskine-arm** [comment]
@danh-arm There were two problems: a test case was failing, and the test script was incorrectly reporting it as having passed (but the logs showed that it had failed). I remember we fixed that failing test case a long time ago, so you wouldn't see the failure happening anymore. But I don't see any obvious sign that if that test failed again, the failure would now be reported correctly.

**Comment by danh-arm** [comment]
OK thanks @gilles-peskine-arm . Given the test is no longer failing, is the cost of fixing worth the benefit? To fix this properly would require writing a regression test that deliberately triggers the failure and checks it fails gracefully, which doesn't sound trivial. Maybe we can lower the bar for this one and just check locally if the fix works?

**Comment by gilles-peskine-arm** [comment]
> Maybe we can lower the bar for this one and just check locally if the fix works?

Yes, that's what we usually do for bugs in tests. We don't have tests for tests.

**Comment by RonEld** [comment]
@danh-arm 
I have rebased and amended the commit message.

To be honest, I don't have much capacity for doing rework on my PRs, so I would appreciate if this PR will get merged soon.

File: programs/ssl/ssl_server2.c (hunk 1 above)
Comment (gilles-peskine-arm) [reply programs/ssl/ssl_server2.c:3721]: Thanks for updating. There's a word (at least) missing from the commit message:

> When `ssl_server2` export key functionality,
> don't exit the server (…)

Did you mean “When the `ssl_server2` export key functionality fails,”? Or something else?

Comment (gilles-peskine-arm) [reply programs/ssl/ssl_server2.c:3721]: While you're at it, in the commit message:

* “sll-opts.sh” → “ssl-opt.sh”
* “for verifying export suceeded” → “to verify if the export succeeded”

**Comment by RonEld** [comment]
@gilles-peskine-arm I have amended the commit message
